src/CommunityToolkit.Maui/Layouts/StateLayout/StateLayoutController.shared.cs

```diff
@@ -0,0 +1,285 @@
+	/// <param name="customState"></param>
+	/// <param name="animate"></param>
+	/// <exception cref="ArgumentException"></exception>
+	public async Task SwitchToTemplate(LayoutState state, string? customState, bool animate)
+	{
+		var layout = GetLayout();
+		var token = RebuildAnimationTokenSource(layout);
+
+		// Put the original content somewhere where we can restore it.
+		if (previousState is LayoutState.None)
+		{
+			originalContent = new List<View>();
+
+			foreach (var item in layout.Children)
+			{
+				originalContent.Add((View)item);
+			}
+		}
+
+		var view = GetViewForState(state, customState);
+
+		if (view is not null)
+		{
+			previousState = state;
+
+			await FadeLayoutChildren(layout, animate, true, token);
+
+			if (token.IsCancellationRequested)
+			{
+				return;
+			}
+
+
+			layout.Children.Clear();
+
+			var repeatCount = GetRepeatCount(state, customState);
+			var template = GetTemplate(state, customState);
+
+			if (template is not null)
+			{
+				// We have a template we can use.
+				var items = new List<int>(repeatCount);
+
+				for (var i = 0; i < repeatCount; i++)
+				{
+					items.Add(i);
+				}
+
+				// We create a VerticalStackLayout to stack repeating items.
+				// It takes VerticalOptions and HorizontalOptions from the
+				// StateView to allow for more control over how it layouts.
+				var s = new VerticalStackLayout
+				{
+					Opacity = animate ? 0 : 1,
+					VerticalOptions = view.VerticalOptions,
+					HorizontalOptions = view.HorizontalOptions
+				};
+
+				// If the layout we're applying StateLayout to is a Grid,
+				// we want to have the StateLayout span the entire Grid surface.
+				// Otherwise it would just end up in row 0 : column 0.
+				if (layout is Grid grid)
+				{
+					if (grid.RowDefinitions.Count > 0)
+					{
+						Grid.SetRowSpan(s, grid.RowDefinitions.Count);
+					}
+
+					if (grid.ColumnDefinitions.Count > 0)
+					{
+						Grid.SetColumnSpan(s, grid.ColumnDefinitions.Count);
+					}
+
+				}
+
+				BindableLayout.SetItemTemplate(s, template);
+				BindableLayout.SetItemsSource(s, items);
+
+				layout.Children.Add(s);
+			}
+			else
+			{
+				if (repeatCount > 1)
+				{
+					throw new ArgumentException("Please use a Template instead of directly defining content when using a RepeatCount > 1.");
+				}
+
+
+				// No template, so we use the children of the StateView.
+				// We create a VerticalStackLayout to stack repeating items.
+				// It takes VerticalOptions and HorizontalOptions from the
+				// StateView to allow for more control over how it layouts.
+				var s = new VerticalStackLayout
+				{
+					Opacity = animate ? 0 : 1,
+					VerticalOptions = view.VerticalOptions,
+					HorizontalOptions = view.HorizontalOptions
+				};
+
+				// If the layout we're applying StateLayout to is a Grid,
+				// we want to have the StateLayout span the entire Grid surface.
+				// Otherwise it would just end up in row 0 : column 0.
+				if (layout is Grid grid)
```

Comment:
seems like this code is the same as above. please extract it to remove duplication